invoice/src/main/java/org/killbill/billing/invoice/model/InvoiceItemFactory.java

```diff
@@ -141,7 +141,14 @@ public static InvoiceItem fromModelDaoWithCatalog(final InvoiceItemModelDao invo
             if (computePrettyName) {
 
                 if (planName != null) {
-                    final Plan plan = catalog.findPlan(planName, createdDate);
+
+                    // We don't know the creation date of the said subscription, all we know is the time
+                    // of the current event. By specifying KILLBILL_GENESIS, we should at least find a Plan
+                    // and most likely the correct one since we start from the transitionDate and move backward
+                    // Worst case: This is the wrong plan and the prettyName was updated across version, oh well..
+                    //
+                    final DateTime KILLBILL_GENESIS = new DateTime(2019, 10, 28, 0, 0);
+                    final Plan plan = catalog.findPlan(planName, transitionDate, KILLBILL_GENESIS);
```

Comment:
Should we do the same fix in the Analytics plugin to keep the logic in sync (https://github.com/killbill/killbill-analytics-plugin/blob/f93f95a56c2068037fae6821d78c6df558f4be1e/src/main/java/org/killbill/billing/plugin/analytics/dao/factory/BusinessFactoryBase.java#L331)?

Reply:
There is a tack i had opened to track the work on the analytics side: `https://github.com/killbill/killbill-analytics-plugin/issues/99`. I don't know if this is the right fix for this.

This fix was a great way to go around the exception but does not guarantee 100% we are pulling the right version. For a *pretty* string this is good enough but for any price calculation, i am not so sure.